haystack/utils/experiment_tracking.py

```diff
@@ -0,0 +1,187 @@
+from abc import ABC, abstractmethod
+import logging
+from pathlib import Path
+from typing import Any, Dict, Union
+import mlflow
+from requests.exceptions import ConnectionError
+
+from haystack.utils.environment import get_or_create_env_meta_data
+
+
+logger = logging.getLogger(__name__)
+
+
+def flatten_dict(dict_to_flatten: dict, prefix: str = ""):
```

Comment:
We have a similar `flatten_list` in `haystack/modeling/utils.py`. Maybe we could move both to `haystack/utils` somewhere?